runtime/major_gc.c

```diff
@@ -382,6 +384,8 @@ static void start_cycle (void)
   caml_gc_message (0x01, "Starting new major GC cycle\n");
   caml_darken_all_roots_start ();
   caml_gc_phase = Phase_mark;
+  marked_words = 0;
```

Comment:
This line should be two lines earlier: currently, it ignores the contribution to `marked_words` made by `caml_darken_all_roots_start`